docs/developer.md

```diff
@@ -21,3 +21,35 @@ The PHP code should all adhere to [PSR-2](https://www.php-fig.org/psr/psr-2/).
 To test the codestyle you can run `make phpcs`.
 
 For linting JavaScript, a [jshint file](https://github.com/nextcloud/news/blob/master/js/.jshintrc) is used that is run before compiling the JavaScript.
+
+## General Developer setup
+Check the Nextcloud [documentation](https://docs.nextcloud.com/server/latest/developer_manual/getting_started/devenv.html) to learn how to setup a developer environment, alternatively to a proper web server you can also use the [builtin php server](https://www.php.net/manual/en/features.commandline.webserver.php) on demand, it is enough for development purposes.
+
+When your setup is running, clone the news repository in the `apps/` directory inside the server.
+
+Change into the news directory and run make to build the app, you will need php, composer, node, npm and maybe more.
+
+Now you can basically use the news app and test your changes.
+
+## Running Integration tests locally
+We use [bats](https://bats-core.readthedocs.io/en/stable/) to run integration tests against the API and the cli.
+
+Check how to install bats on your system in the [official documentation](https://bats-core.readthedocs.io/en/stable/installation.html).
+
+You also need to pull the submodules of the news repo.
+```bash
+git submodules update --init
+```
+
+The cli tests expect that the feeds are reachable at `http://localhost:8090`, to achieve that you can use the [builtin php server](https://www.php.net/manual/en/features.commandline.webserver.php).
+
+Change into the `tests/test_helpers/feeds` directory and execute `php -S localhost:8090` you can also run it in the background like this `php -S localhost:8090 &`.
```

Comment:
Can we make this a `make` target?

Reply:
I think that would be possible but only with assumptions about the environment.

Locally I have the news repo cloned into the apps directory of the server repo.
I think on the actions runner we have the server with a build of the app in the apps directory but the repo itself is outside.

And I'm not so interested in building that, I just wanted to document the current status.

Reply:
Do you mind keeping it open till I can add it then?

Reply:
Yea that's fine :)